kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -203,11 +203,34 @@ public CloseableIterator<FilteredColumnarBatch> getAddFilesAsColumnarBatches(
   protected Tuple2<Protocol, Metadata> loadTableProtocolAndMetadata(
       Engine engine, Optional<SnapshotHint> snapshotHint, long snapshotVersion) {
 
-    // Exit early if the hint already has the info we need
+    // If the snapshot hint is provided and matches the expected version, load the P&M from version hint.
+    // If not, e.g. for the case of time travel, skip use version hint and fallback to CRC file or log replay.
```

Comment:
Err, I don't want this to imply that time travel is the _only_ case where the snapshot hint isn't used. Also, we aren't really "skipping " the version hint, we are just performing log replay, which _still_ may end up using the version hint.

Let's just revert this back to `// Exit early if the hint already has the info we need`.

Reply:
Done